lib/binary_to_decimal.rb

```diff
@@ -2,8 +2,14 @@
 # The array is randomly filled with 0’s and 1’s.
 # The most significant bit is at index 0.
 # The least significant bit is at index 7.
-# Calculate  and return the decimal value for this binary number using
-# the algorithm you devised in class.
-def binary_to_decimal(binary_array)
-  raise NotImplementedError
+# Calculate  and return the decimal value for this binary number using the algorithm you devised in class.
+
+def binary_to_decimal(binary_array)  
+  decimal_num = 0
+  binary_array.each_with_index do |bit_num, i|
+    if bit_num == 1
+      decimal_num += 2**(7-i)
```

Comment:
I suggest using a constant or `binary_array.length -1` in place of 7.  That just makes the code a bit more flexible.